src/dstack/_internal/server/background/tasks/process_idle_volumes.py

```diff
@@ -0,0 +1,163 @@
+            .options(selectinload(VolumeModel.project))
+            .options(selectinload(VolumeModel.user))
+            .options(selectinload(VolumeModel.attachments))
+            .execution_options(populate_existing=True)
+        )
+        volumes = list(res.unique().scalars().all())
+
+        try:
+            to_delete = []
+            for volume in volumes:
+                if _should_delete_volume(volume):
+                    to_delete.append(volume)
+
+            if to_delete:
+                await _delete_idle_volumes(session, to_delete)
+
+        finally:
+            lockset.difference_update(volume_ids)
+
+
+def _should_delete_volume(volume: VolumeModel) -> bool:
+    config = get_volume_configuration(volume)
+
+    if not config.auto_cleanup_duration:
+        return False
+
+    if isinstance(config.auto_cleanup_duration, int) and config.auto_cleanup_duration < 0:
+        return False
+
+    duration_seconds = parse_duration(config.auto_cleanup_duration)
+    if not duration_seconds or duration_seconds <= 0:
+        return False
+
+    if volume.attachments:
+        return False
+
+    idle_time = _get_idle_time(volume)
+    threshold = datetime.timedelta(seconds=duration_seconds)
+
+    if idle_time > threshold:
+        logger.info(
+            "Deleting idle volume %s (idle %.1fh)", volume.name, idle_time.total_seconds() / 3600
+        )
+        return True
+
+    return False
+
+
+def _get_idle_time(volume: VolumeModel) -> datetime.timedelta:
+    last_used = volume.last_job_processed_at or volume.created_at
+    last_used_utc = last_used.replace(tzinfo=datetime.timezone.utc)
+    now = get_current_datetime()
+
+    idle_time = now - last_used_utc
+    return max(idle_time, datetime.timedelta(0))
+
+
+async def _delete_idle_volumes(session: AsyncSession, volumes: List[VolumeModel]):
+    """Delete idle volumes from cloud providers and mark as deleted in database."""
+    for volume_model in volumes:
+        try:
+            await _delete_volume_from_cloud(session, volume_model)
+        except Exception:
+            logger.exception("Error when deleting volume %s from cloud", volume_model.name)
+        try:
+            await session.execute(
+                update(VolumeModel)
+                .where(VolumeModel.id == volume_model.id)
+                .values(
+                    deleted=True,
+                    deleted_at=get_current_datetime(),
+                )
+            )
+            logger.info("Deleted idle volume %s", volume_model.name)
+        except Exception:
+            logger.exception("Failed to mark volume %s as deleted in database", volume_model.name)
+
+    await session.commit()
+
+
+async def _delete_volume_from_cloud(session: AsyncSession, volume_model: VolumeModel):
+    """Delete volume from cloud provider. Based on volumes.py:_delete_volume"""
+    volume = volume_model_to_volume(volume_model)
+
+    if volume.external:
+        return
```

Comment:
Maybe disallow creating external volumes with auto_cleanup_duration if it's not supported?